prow/sidecar/run.go

```diff
@@ -134,7 +137,7 @@ func (o Options) Run(ctx context.Context) (int, error) {
 
 	buildLogs := logReaders(entries)
 	metadata := combineMetadata(entries)
-	return failures, o.doUpload(context.Background(), spec, passed, aborted, metadata, buildLogs)
+	return failures, o.doUpload(context.Background(), spec, passed, aborted, metadata, buildLogs, logFilePath)
```

Comment:
Note that `doUpload` can be called twice. We need to handle that case gracefully. I think if my other suggestions are followed the file upload will be idempotent.

Reply:
yeah, you're right i see that. ill address all other concerns  and suggestions then